google/cloud/spanner_v1/snapshot.py

```diff
@@ -969,14 +1023,24 @@ def begin(self):
             observability_options=getattr(database, "observability_options", None),
             metadata=metadata,
         ), MetricsCapture():
-            method = functools.partial(
-                api.begin_transaction,
-                session=self._session.name,
-                options=txn_selector.begin,
-                metadata=metadata,
-            )
+            nth_request = getattr(database, "_next_nth_request", 0)
+            counters = dict(attempt=0)
```

Comment:
Is there a specific reason that this uses a dictionary with an attempt, instead of an atomic counter?

Reply:
It is a legacy artifact, fixed and thank you!